packages/holocron/src/externalRegistry.js

```diff
@@ -15,42 +15,117 @@
+ *         semanticRange: {string}[range of expected external],
+ *         integrity: {string}[hash value of module code]
+ *      }
+ *    }
+ * }
+ * @type { Immutable.Map } registry
+ * @param {string} registry.[moduleName].[externalName].name external name
+ * @param {string} registry.[moduleName].[externalName].version external version
+ * @param {string} registry.[moduleName].[externalName].semanticRange accepted semantic range
+ * @param {string} registry.[moduleName].[externalName].integrity hash value of fallback external
+ */
 let requiredModuleExternals = iMap();
 
+/**
+ * returns registry of registered externals
+ * @returns { object } - registered external registry
+ */
 function getRegisteredExternals() {
   return registeredExternals.toJS();
 }
 
+/**
+ * Validate if given range is satisfied by the provided version
+ * @param {string} {providedVersion - version to validate against given range
+ * @param {string} requestedRange} - semantic range
+ * @returns {boolean}
+ */
 function validateExternal({ providedVersion, requestedRange }) {
   return semver.satisfies(providedVersion, requestedRange);
 }
 
+/**
+ * Add external to the registered externals registry
+ * @param {string} name name of the external
+ * @param {string} version version of the externals
+ * @param {any} module the externals code
+ */
 function registerExternal({ name, version, module }) {
   registeredExternals = registeredExternals.setIn([name, version], module);
 }
 
+/**
+ * Returns the code of the requested external
+ * @param {any} name name of the external dependency
+ * @param {any} version version of the external dependency
+ * @returns {any} externals code
+ */
 function getExternal({ name, version }) {
   return registeredExternals.getIn([name, version]);
 }
 
+/**
+ * validate if specific version of external dependency has been registered
+ * @param {string} name
+ * @param {string} version
+ * @returns {boolean}
+ */
 function hasExternalLoaded({ name, version }) {
   return registeredExternals.hasIn([name, version]);
 }
 
+/**
+ * Remove registered required externals for given module
+ * @param {string} moduleName
+ */
 function clearModulesRequiredExternals(moduleName) {
   requiredModuleExternals = requiredModuleExternals.delete(moduleName);
 }
 
-function setModulesRequiredExternals({ moduleName, externals }) {
+/**
+ * Register missing external dependencies that a module requires to load
+ * @param {string} moduleName name of the holocron module
+ * @param {object} externals externals to be added
+ * @param {object} externals.[externalName]
+ * @param {string} externals.[externalName].name external name
+ * @param {string} externals.[externalName].version external version
+ * @param {string} externals.[externalName].semanticRange semantic range module will accept
+ * @param {string} externals.[externalName].integrity hash value of fallback external
+ */
+function setModulesRequiredExternals({ moduleName, externals = {} }) {
```

Comment:
If externals are undefined, should we be setting anything at all?

Reply:
i thought the same, landed on if a moduleName is provided it would be expected to be entered into the registry.

Reply:
which is why i think it could be empty.